service-base/src/main/java/org/eclipse/hono/service/plan/PrometheusBasedResourceLimitChecks.java

```diff
@@ -132,21 +204,26 @@ public PrometheusBasedResourceLimitChecks(final WebClient webClient,
         final DataVolume dataVolumeConfig = tenant.getResourceLimits().getDataVolume();
         final long maxBytes = dataVolumeConfig.getMaxBytes();
         final Instant effectiveSince = dataVolumeConfig.getEffectiveSince();
+        final PeriodMode periodMode = PeriodMode.from(dataVolumeConfig.getPeriod().getMode());
         final long periodInDays = Optional.ofNullable(dataVolumeConfig.getPeriod())
                 .map(DataVolumePeriod::getNoOfDays)
                 .orElse(0);
 
-        log.trace("message limit config for tenant [{}] are [{}:{}, {}:{}, {}:{}]", tenant.getTenantId(),
+        log.trace("message limit config for tenant [{}] are [{}:{}, {}:{}, {}:{}, {}:{}]", tenant.getTenantId(),
                 TenantConstants.FIELD_MAX_BYTES, maxBytes,
                 TenantConstants.FIELD_EFFECTIVE_SINCE, effectiveSince,
+                TenantConstants.FIELD_PERIOD_MODE, periodMode,
                 TenantConstants.FIELD_PERIOD_NO_OF_DAYS, periodInDays);
 
-        if (maxBytes == -1 || effectiveSince == null || periodInDays <= 0 || payloadSize <= 0) {
+        if (maxBytes == -1 || effectiveSince == null || PeriodMode.UNKNOWN.equals(periodMode) || payloadSize <= 0) {
             return Future.succeededFuture(Boolean.FALSE);
         } else {
-            final long dataUsagePeriod = calculateDataUsagePeriod(effectiveSince, periodInDays);
+            final long dataUsagePeriod = calculateDataUsagePeriod(
+                    OffsetDateTime.ofInstant(effectiveSince, ZoneOffset.UTC), periodMode, periodInDays);
+            final long allowedMaxBytes = calculateDataVolume(OffsetDateTime.ofInstant(effectiveSince, ZoneOffset.UTC),
+                    periodMode, maxBytes);
 
-            if (dataUsagePeriod <= 0) {
+            if (dataUsagePeriod <= 0 || allowedMaxBytes <= 0) {
```

Comment:
why isn't the limit reached if allowed max bytes is 0? To me, the value suggests: no data is allowed at all. Or am I mistaken?

Reply:
If it is zero then it is interpreted as no data is allowed and this check I will remove it.